src/components/Disputes/Import.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable  */
+import React, { useState, useEffect } from 'react'
+import { SidePanel, Button, DropDown, TextInput, textStyle } from '@aragon/ui'
+import { Octokit } from '@octokit/rest'
+import { useWallet } from 'use-wallet'
+
+function Import() {
+  const wallet = useWallet()
+
+  const [USERNAME, SETUSERNAME] = useState(null)
+
+  const [opened, setOpened] = useState(false)
+
+  const [list, setList] = useState([])
+
+  const TOKEN = localStorage.getItem('ACCESS TOKEN')
+
+  const octokit = new Octokit({
+    auth: TOKEN,
+  })
+
+  const [publicRepos, setPublicRepos] = useState(0)
+  async function userInfo() {
+    const { data } = await octokit.request('/user')
+    console.log('userinfo', data)
+    let num = data.public_repos
+    setPublicRepos(num - 1)
+    return data
+  }
+  async function listUserRepos() {
+    console.log(localStorage.getItem('ACCESS TOKEN'))
+    setOpened(true)
+
+    let repoList = []
+    const user = await userInfo()
+    for (let page = 1; page <= Math.ceil(user.public_repos / 100); page++) {
+      const { data } = await octokit.repos.listForAuthenticatedUser({
+        page: page,
+      })
+      console.log('data', data)
+      let x = 0
+      let arr = []
```

Comment:
You need to move this `arr` outside the `for` loop, otherwise, the array will be reinitialized for every page fetch. In my case I have 130 public repos but it shows only 30 on the list.

Reply:
Fixed